codespan-reporting/src/term/renderer.rs

```diff
@@ -127,6 +129,10 @@ impl<'writer, 'config> Renderer<'writer, 'config> {
         &self.config.styles
     }
 
+    pub fn context_lines(&self) -> usize {
+        self.config.context_lines
+    }
```

Comment:
I'm guessing this is redundant now?

Reply:
No, this is the only way to access the configuration from within `RichDiagnostic::render`.

Reply:
Ahhh, interesting! Yeah it seems kind of annoying that we need to propagate information back to the view like this. Not sure what can be done about it though, and the separation between the view and the renderer is not the greatest right now if I'm honest - ie. we do a bit of layout-ish computation in the renderer.

Reply:
But I think this problem is outside the scope of this PR, right?

Reply:
Yeah, agreed. Sorry I should have been more clear on that. 😓